packages/frontend/editor-ui/src/features/dataStore/components/dataGrid/ColumnHeader.vue

```diff
@@ -0,0 +1,118 @@
+<script setup lang="ts">
+import type { IHeaderParams } from 'ag-grid-community';
+import { useDataStoreTypes } from '@/features/dataStore/composables/useDataStoreTypes';
+import { ref, computed } from 'vue';
+import { useI18n } from '@n8n/i18n';
+import { isAGGridCellType } from '@/features/dataStore/types';
+
+type HeaderParamsWithDelete = IHeaderParams & {
+	onDelete: (columnId: string) => void;
+};
+
+const props = defineProps<{
+	params: HeaderParamsWithDelete;
+}>();
+
+const { getIconForType, mapToDataStoreColumnType } = useDataStoreTypes();
+const i18n = useI18n();
+
+const isHovered = ref(false);
+const isDropdownOpen = ref(false);
+
+const enum ItemAction {
+	Delete = 'delete',
+}
+
+const onItemClick = (action: string) => {
+	if (action === (ItemAction.Delete as string)) {
+		props.params.onDelete(props.params.column.getColId());
+	}
+};
+
+const onMouseEnter = () => {
+	isHovered.value = true;
+};
+
+const onMouseLeave = () => {
+	isHovered.value = false;
+};
+
+const onDropdownVisibleChange = (visible: boolean) => {
+	isDropdownOpen.value = visible;
+};
+
+const isDropdownVisible = computed(() => {
+	return isHovered.value || isDropdownOpen.value;
+});
+
+const typeIcon = computed(() => {
+	const cellDataType = props.params.column.getColDef().cellDataType;
+	if (!isAGGridCellType(cellDataType)) {
+		return null;
+	}
+	return getIconForType(mapToDataStoreColumnType(cellDataType));
+});
+</script>
+<template>
+	<div
+		class="ag-header-cell-label data-store-column-header-wrapper"
+		data-test-id="data-store-column-header"
+		@mouseenter="onMouseEnter"
+		@mouseleave="onMouseLeave"
+	>
+		<div class="data-store-column-header-icon-wrapper">
+			<N8nIcon v-if="typeIcon" :icon="typeIcon" />
+			<span class="ag-header-cell-text" data-test-id="data-store-column-header-text">{{
+				props.params.displayName
+			}}</span>
+		</div>
+		<N8nActionDropdown
+			v-show="isDropdownVisible"
+			data-test-id="data-store-column-header-actions"
+			:items="[
+				{
+					id: ItemAction.Delete,
+					label: i18n.baseText('dataStore.columnActions.delete'),
+					icon: 'trash-2',
+					customClass: 'data-store-column-header-action-item',
+				},
+			]"
+			:placement="'bottom-start'"
+			:activator-icon="'ellipsis'"
+			@select="onItemClick"
+			@visible-change="onDropdownVisibleChange"
+		/>
+	</div>
+</template>
+
+<style lang="scss">
+// TODO: neither scoped nor module works here. Is there a way to resolve this?
```

Comment:
The only way I managed to do this is by defining classes in `DataStoreTable` component and passing `$style` as a prop to the header ([here](https://github.com/n8n-io/n8n/compare/ado-3901-data-store-column-headers...ag-grid-css-modules)). Not sure I like this but it can be a way if we really want css modules here.